app/controllers/feedbacks_controller.rb

```diff
@@ -0,0 +1,74 @@
+class FeedbacksController < ApplicationController
+  before_action :set_feedback, only: [:show, :edit, :update, :destroy]
+  before_action :set_feedbackable, only: [:new, :create]
+
+  def index
+    authorize Feedback
+    @feedbacks = policy_scope(Feedback)
+    if params[:group_offer_id]
+      @feedbacks = @feedbacks.where(feedbackable_id: params[:group_offer_id])
+      @volunteer = current_user.volunteer || @feedbacks.first.volunteer
+    elsif params[:assignment_id]
+      @feedbacks = @feedbacks.where(feedbackable_id: params[:assignment_id])
+      @volunteer = Assignment.find(params[:assignment_id]).volunteer
+    elsif params[:volunteer_id]
+      @feedbacks = @feedbacks.where(volunteer_id: params[:volunteer_id])
+      @volunteer = Volunteer.find(params[:volunteer_id])
+    end
+  end
+
+  def show; end
+
+  def new
+    @feedback = Feedback.new(feedbackable: @feedbackable, volunteer: @volunteer,
+      author: current_user)
+    authorize @feedback
+  end
+
+  def edit; end
+
+  def create
+    @feedback = Feedback.new(feedback_params.merge(author_id: current_user.id))
+    authorize @feedback
+    if @feedback.save
+      redirect_to url_for([@feedback.volunteer, @feedback]), make_notice
```

Comment:
Not sure if `@feedback.volunteer` is correct for the cases where the feedbackable model isn't a volunteer. Maybe use `@feedback.feedbackable` as well?

Reply:
feedbackable are only `:group_offers` and `:assignments`. But Feedbacks have a relation to `:volunteer`.

Reply:
Ah ok, in that case ignore my comment :)